pyresample/future/spherical/arc.py

```diff
@@ -0,0 +1,209 @@
+        """Overwrite a Arc deprecated function. Inherited from Arc class."""
+        raise ValueError("This function is deprecated.")
+
+    def intersections(self, other_arc):
+        """Overwritea Arc deprecated function. Inherited from Arc class."""
+        raise ValueError("'SArc.intersections' is deprecated. Use '_great_circle_intersections' instead.")
+
+    def _great_circle_intersections(self, other_arc):
+        """Compute the intersections points of the greats circles over which the arcs lies.
+
+        A great circle divides the sphere in two equal hemispheres.
+        """
+        end_lon = self.end.lon
+        other_end_lon = other_arc.end.lon
+
+        if self.end.lon - self.start.lon > np.pi:
+            end_lon -= 2 * np.pi
+        if other_arc.end.lon - other_arc.start.lon > np.pi:
+            other_end_lon -= 2 * np.pi
+        if self.end.lon - self.start.lon < -np.pi:
+            end_lon += 2 * np.pi
+        if other_arc.end.lon - other_arc.start.lon < -np.pi:
+            other_end_lon += 2 * np.pi
+
+        end_point = SPoint(end_lon, self.end.lat)
+        other_end_point = SPoint(other_end_lon, other_arc.end.lat)
+
+        ea_ = self.start.cross2cart(end_point).normalize()
+        eb_ = other_arc.start.cross2cart(other_end_point).normalize()
+
+        cross = ea_.cross(eb_)
+        lat = np.arctan2(cross.cart[2],
+                         np.sqrt(cross.cart[0] ** 2 + cross.cart[1] ** 2))
+        lon = np.arctan2(cross.cart[1], cross.cart[0])
+
+        return (SPoint(lon, lat), SPoint(lon + np.pi, -lat))
+
+    def intersection(self, other_arc):
+        """Overwrite a Arc deprecated function. Inherited from Arc class."""
+        raise ValueError("'SArc.intersection' is deprecated. Use 'intersection_point' instead.")
+
+    def intersection_point(self, other_arc):
+        """Compute the intersection point between two arcs.
+
+        If arc and *other_arc* intersect, it returns the intersection SPoint.
+        If arc and *other_arc* does not intersect, it returns None.
+        If same arc (also same direction), it returns None.
+        """
+        # If same arc (same direction), return None
+        if self == other_arc:
+            return None
+
+        great_circles_intersection_spoints = self._great_circle_intersections(other_arc)
+
+        for spoint in great_circles_intersection_spoints:
+            a = self.start
+            b = self.end
+            c = other_arc.start
+            d = other_arc.end
+
+            ab_dist = a.hdistance(b)
+            cd_dist = c.hdistance(d)
+            ap_dist = a.hdistance(spoint)
+            bp_dist = b.hdistance(spoint)
+            cp_dist = c.hdistance(spoint)
+            dp_dist = d.hdistance(spoint)
+
+            if (((spoint in (a, b)) or (abs(ap_dist + bp_dist - ab_dist) < EPSILON)) and
+                    ((spoint in (c, d)) or (abs(cp_dist + dp_dist - cd_dist) < EPSILON))):
+                return spoint
```

Comment:
Could be clean up this code? Perhaps remove the `a/b/c/d` naming with longer more readable names? More importantly, can the `if` conditions be broken up into variables to describe what is being checked? For example:

```python
point_is_this_arc = spoint in (self.start, self.end)
point_is_other_arc = spoint in (other_arc.start, other_arc.end)
point_is_on_this_arc_segment = abs(...) < EPSILON  # (or better names for whatever this calculation is)
point_is_on_other_arc_segment = ...
point_is_on_this_arc = point_is_this_arc_end or point_is_on_this_arc_segment
point_is_on_other_arc_segment = point_is_other_arc_end or point_is_on_other_arc_segment
if point_is_on_this_arc and point_is_on_other_arc:
    return spoint
````